File: Que 1/C/Patterns.c
#include<stdio.h>

void Right_angled_triangle(int);
void Isosceles_triangle(int);
void kite(int);
void half_kite(int);
void x(int);
void repeat(char, int);
void menu();


void repeat(char c, int num){
    printf(" ");
    for(int i=0; i<num; i++){
        printf("%c", c);
    }
}

void menu(){
    
    char b = '~';
    int a = 37, choice, size;

    repeat(b, a);
    printf("\n");

    printf(" *\t\tMENU\t\t     *\n");
    
    repeat(b, a);
    printf("\n");
    
    printf(" *  1. Right-angled Triangle\t     *\n *  2. Isosceles Triangle\t     *\n *  3. Kite\t\t\t     *\n *  4. Half Kite\t\t     *\n *  5. X\t\t\t     *\n");
    
    repeat(b, a);
    printf("\n");
    
    printf("\tEnter your Choice : ");
    scanf("%d", &choice);
    
    repeat(b, a);
    printf("\n");
    
    if ((choice == 1) || (choice == 2) || (choice == 3) || (choice == 4) || (choice == 5)){

        printf("\tEnter the Size : ");
        scanf("%d", &size);
        
        repeat(b, a);
        printf("\n");
        
        switch (choice){

            case 1:
                Right_angled_triangle(size);
                break;

            case 2:
                if(size % 2 == 0){
                    Isosceles_triangle(size);
                }else{
                    printf("     Size should be even number.");
                }
                break;
                
            case 3:
                if(size % 2 == 0){
                    kite(size);
                }else{
                    printf("     Size should be even number.");
                }
                break;
            
            case 4:
                half_kite(size);
                break;

            case 5:
                if(size % 2 == 0){
                    x(size);
                }else{
                    printf("     Size should be even number.");
                }
                break;
                
            default:
                printf("\t     Try Again!!!");
        }
    }else{
        printf("\t      Try Again!!!");
    }
    
    repeat(b, a);
    printf("\n");

}

void x(int n){
    char s = ' ';
    int count = n * 2;
        for(int i=1; i<=n; i++){
            repeat(s, i);
            for(int j=1; j<count; j++){
                printf("%d",j);
            }
            printf("\n");
            count -= 2;
        }

        count = 2;
        for(int i = n - 1; i>0; i--){
            repeat(s, i);
            count += 2;
            for(int j=1; j<count; j++){
                printf("%d",j);
            }
            printf("\n");
        }
    }


void half_kite(int n){
    int count = 1;
        for(int i=0; i<n; i++){
            for(int j=1; j<=count; j++){
                printf(" %d ",j);
            }
            count += 1;
            printf("\n");
        }
    count = n;
        for(int i=0; i<n-1; i++){
            for(int j=1; j<count; j++){
                printf(" %d ",j);
            }
            count -= 1;
            printf("\n");
        }
}

void kite(int n){
    Isosceles_triangle(n);
    char s = ' ';
    int count = n + 1;
        for(int i=2; i<=n; i++){
            repeat(s, i);
            for(int j=1; j<=count; j++){
                printf("%d",j);
            }
            count -= 2;
            printf("\n");
        }

}

void Isosceles_triangle(int n) {
    char s = ' ';
    int count = 1;
    for(int i=n; i>0; i--){
        repeat(s, i);
        for(int j=1; j<=count; j++){
            printf("%d",j);
        }
        count = count + 2;
        printf("\n");
    }

}

void Right_angled_triangle(int n){
    for(int i=1; i<=n; i++){
            printf("%s", "  ");
            for(int j=1; j<=i; j++){
                printf("%d ", j);
            }
            printf("\n");
        }
}

void main(){
    int n = 4, a = 1;
    char choice;

    while(a == 1){
        menu();
        printf("   Do you want to continue(Y/N) : ");
        scanf("%s", &choice);
        
        if (choice == 'y' || choice == 'Y'){
            a = 1;
        }else 
            a = 0;
        repeat('~', 37);
        printf("\n");
    }   

}

File: Que 1/Python/Patterns.py

#_author__ : 'Devanshu'

class Patterns:

	def __init__(self):
		self.a = 37
		self.b = "-"
		self.choice = 0 
		
	def main(self):
		print(self.b * self.a)
		print("|\t\t MENU\t\t    |")
		print(self.b * self.a)
		print(
"""|  1. Right-angled triangle         |
|  2. Isosceles triangle            |
|  3. Kite                          |
|  4. Half Kite                     |
|  5. X                             |"""
		)
		print(self.b * self.a)

		self.choice = input("\tEnter your choice : ")
		print(self.b * self.a)

		try:
			c = int(self.choice)
			size = int(input("\tEnter the Size : "))
			self.n = size
			print(self.b * self.a)

			if int(self.choice) == 1:
				self.right_angle_triangle()

			elif int(self.choice) == 2:
				if int(self.n) % 2 == 0:
					self.Isosceles_triangle()
				else:
					print("|    Size should be even number.  |")
			
			elif int(self.choice) == 3:
				if int(self.n) % 2 == 0:
					self.kite()
				else:
					print("|    Size should be even number.  |")

			elif int(self.choice) == 4:
				self.half_kite()
			
			elif int(self.choice) == 5:
				if int(self.n) % 2 == 0:
					self.x()
				else:
					print("|    Size should be even number.  |")  
			
			else:
				print("|\t  Try again !!!  |")
		except:
			print("|\t  Try again !!!  |")

	def right_angle_triangle(self):
		for i in range(1, self.n + 1):
			print("   ", end="")
			for j in range(1, i + 1):
				print(j, " ", end='')
			print("")

	def Isosceles_triangle(self):
		count = 1
		for i in range(self.n, 0, -1):
			print(" "*i, end="")
			for j in range(1, count+1):
				print(j, end="")
			count += 2
			print("")

	def kite(self):
		self.Isosceles_triangle()
		count =  self.n + 1
		for i in range(2, self.n+1):
			print(" "*i, end="")
			for j in range(1, count+1):
				print(j, end="")
			count -= 2
			print("")

	def half_kite(self):
		count = 1
		for i in range(self.n):
			for j in range(1, count+1):
				print(" ", j, " ", end="")
			count += 1
			print("")
		count = self.n
		for i in range(self.n-1):
			for j in range(1, count):
				print(" ", j, " ", end="")
			count -= 1
			print("")

	def x(self):
		count = self.n * 2
		for i in range(1, self.n+1):
			print(" "*i, end="")
			for j in range(1, count):
				print(j, end="")
			print("")
			count -= 2
		
		count = 2
		for i in range(self.n-1, 0, -1):
			print(" "*i, end="")
			count += 2
			for j in range(1, count):
				print(j, end="")
			print("")
	

if __name__ == "__main__":
	a = Patterns()
	a.main()
	#print("-"*35)
	while True:
		print("-"*37)
		choice = input("  Do you want to continue (Y/N) : ")
		if choice.lower() == "y":
			print("-"*37)
			a.main()
		else:
			print("-"*37)
			break